datumaro/plugins/synthetic_data/utils.py

```diff
@@ -0,0 +1,198 @@
+                a, b, c, d, e, f = func_params
+                if rand <= select_func:
+                    next_x = prev_x * a + prev_y * b + e
+                    next_y = prev_x * c + prev_y * d + f
+                    break
+
+            self.xs.append(next_x)
+            self.ys.append(next_y)
+            prev_x = next_x
+            prev_y = next_y
+
+    @staticmethod
+    def process_nans(data):
+        nan_index = np.nonzero(np.isnan(data))
+        extend = np.array(range(nan_index[0][0] - 100, nan_index[0][0]))
+        delete_row = np.append(extend, nan_index)
+        return delete_row
+
+    def rescale(self, image_x, image_y, pad_x, pad_y):
+        if image_x < 2 * pad_x or image_y < 2 * pad_y:
+            raise ValueError(
+                f"Image generation with height < {2 * pad_x} or width < {2 * pad_y} is not supported"
+            )
+
+        xs = np.array(self.xs)
+        ys = np.array(self.ys)
+        if np.any(np.isnan(xs)):
+            delete_row = self.process_nans(xs)
+            xs = np.delete(xs, delete_row, axis=0)
+            ys = np.delete(ys, delete_row, axis=0)
+
+        if np.any(np.isnan(ys)):
+            delete_row = self.process_nans(ys)
+            xs = np.delete(xs, delete_row, axis=0)
+            ys = np.delete(ys, delete_row, axis=0)
+
+        if np.min(xs) < 0.0:
+            xs -= np.min(xs)
+        if np.min(ys) < 0.0:
+            ys -= np.min(ys)
+        xmax, xmin = np.max(xs), np.min(xs)
+        ymax, ymin = np.max(ys), np.min(ys)
+        self.xs = np.uint16(xs / (xmax - xmin + 1e-5) * (image_x - 2 * pad_x) + pad_x)
+        self.ys = np.uint16(ys / (ymax - ymin + 1e-5) * (image_y - 2 * pad_y) + pad_y)
+
+    def draw(self, image_x, image_y, draw_point, pad_x=6, pad_y=6):
+        self.rescale(image_x, image_y, pad_x, pad_y)
+        image = np.zeros((image_x, image_y), dtype=np.uint8)
+        for x, y in zip(self.xs, self.ys):
+            if draw_point:
+                image[x, y] = 127
+            else:
+                mask = "{:09b}".format(self._rng.randint(1, 511))
+                patch = 127 * np.array(list(map(int, list(mask))), dtype=np.uint8).reshape(3, 3)
+                image[x + 1 : x + 4, y + 1 : y + 4] = patch
+
+        return image
+
+
+def download_colorization_model(path):
+    proto_name = "colorization_deploy_v2.prototxt"
+    model_name = "colorization_release_v2.caffemodel"
+    npy_name = "pts_in_hull.npy"
+
+    if not (
+        osp.exists(osp.join(path, proto_name))
+        and osp.exists(osp.join(path, model_name))
+        and osp.exists(osp.join(path, npy_name))
+    ) and not os.access(path, os.W_OK):
+        raise ValueError(
+            "Please provide path where colorization model is located or path to writable directory to save colorization model"
+        )
+
+    if not osp.exists(osp.join(path, proto_name)):
+        log.warning(
+            "Downloading '%s' model for image coloring to '%s'",
```

Comment:
```suggestion
        log.warning(
            "Downloading '%s' model for image coloring to '%s'",
```

I don't think this is a warning. You should use `log.info` instead.